domain/model/service/service.go

```diff
@@ -26,25 +31,71 @@ type State interface {
 // Service defines a service for interacting with the underlying state based
 // information of a model.
 type Service struct {
-	st State
+	st         State
+	toolFinder ToolFinder
+}
+
+// ToolFinder represents a helper for establishing if tools for a specific Juju
+// are available.
+type ToolFinder interface {
+	// HasToolsForVersion will interrogate the tools available in the system and
+	// return nil if tools exist for the specified version. If no tools are
+	// found then a error satisfying [github.com/juju/errors.NotFound] will be
+	// returned. All other errors indicate a problem with parsing the available
+	// tools.
+	HasToolsForVersion(version.Number) error
```

Comment:
Methods called `Has...` are expected to return `bool` or `bool, error`.
So here we should be returning either `true, nil`, `false, nil`, or `false,someerrorotherthannotfound`

Reply:
Agreed